pkg/scaling/scale_handler.go

```diff
@@ -576,81 +587,190 @@ func (h *scaleHandler) getScaledObjectState(ctx context.Context, scaledObject *k
+
+	// when we are using formula, we need to reevaluate if it's active here
+	if scaledObject.IsUsingModifiers() {
+		// we need to reset the activity even if there is an error
+		isScaledObjectActive = false
+		if !isScaledObjectError {
+			activationValue := float64(0)
+			if scaledObject.Spec.Advanced.ScalingModifiers.ActivationTarget != "" {
+				targetValue, err := strconv.ParseFloat(scaledObject.Spec.Advanced.ScalingModifiers.ActivationTarget, 64)
+				if err != nil {
+					return false, true, metricsRecord, fmt.Errorf("scalingModifiers.ActivationTarget parsing error %w", err)
 				}
-				continue
+				activationValue = targetValue
 			}
 
-			metricName := spec.External.Metric.Name
-
-			var latency int64
-			metrics, isMetricActive, latency, err := cache.GetMetricsAndActivityForScaler(ctx, scalerIndex, metricName)
-			if latency != -1 {
-				prommetrics.RecordScalerLatency(scaledObject.Namespace, scaledObject.Name, scalerName, scalerIndex, metricName, float64(latency))
+			for _, metric := range matchingMetrics {
+				value := metric.Value.AsApproximateFloat64()
+				prommetrics.RecordScalerMetric(scaledObject.Namespace, scaledObject.Name, kedav1alpha1.CompositeMetricName, 0, metric.MetricName, value)
+				prommetrics.RecordScalerActive(scaledObject.Namespace, scaledObject.Name, kedav1alpha1.CompositeMetricName, 0, metric.MetricName, value > activationValue)
+				if !isScaledObjectActive {
+					isScaledObjectActive = value > activationValue
+				}
 			}
-			logger.V(1).Info("Getting metrics and activity from scaler", "scaler", scalerName, "metricName", metricName, "metrics", metrics, "activity", isMetricActive, "scalerError", err)
+		}
+	}
 
-			if scalerConfigs[scalerIndex].TriggerUseCachedMetrics {
-				metricsRecord[metricName] = metricscache.MetricsRecord{
-					IsActive:    isMetricActive,
-					Metric:      metrics,
-					ScalerError: err,
-				}
+	// cpu/memory scaler only can scale to zero if there is any other external metric because otherwise
+	// it'll never scale from 0. If all the triggers are only cpu/memory, we enforce the IsActive
+	if len(scaledObject.Spec.Triggers) <= cpuMemCount && !isScaledObjectError {
+		isScaledObjectActive = true
+	}
+	return isScaledObjectActive, isScaledObjectError, metricsRecord, err
+}
+
+// getStateScalerResult is used as paremeter
+// for the function getScalerState. It contains
+// all the needed information for getting the
+// scaler state
+type getStateScalerRequest struct {
+	ctx          context.Context
+	scaler       scalers.Scaler
+	scalerConfig scalers.ScalerConfig
+	scalerIndex  int
+	cache        *cache.ScalersCache
+	logger       logr.Logger
+	scaledObject *kedav1alpha1.ScaledObject
+}
+
+// getStateScalerResult is used as return
+// for the funtion getScalerState. It contains
+// the state of the scaler and all the required
+// info for calculating the ScaledObjectState
+type getStateScalerResult struct {
```

Comment:
```suggestion
type scalerState struct {
```

Reply:
Also, I am not 100% sure that we should use it here. Again it is just used once, right?

Reply:
I will use it for a channel to parallelize it in my next PR